docs/cookbook.rst

```diff
@@ -0,0 +1,3 @@
+=================
+Rasterio Cookbook
```

Comment:
@jdmcbr `.rst` format would be best . In terms of the content, I was thinking bite sized pieces of code (like ~40 lines) for doing discrete chunks of work with rasterio. Not complete workflows but little snippets that are generally useful templates for common tasks. Showing integration with other packages is great but the focus should be rasterio usage. 

I'm picturing something along the lines fof https://pcjericks.github.io/py-gdalogr-cookbook/raster_layers.html - but without  osgeo.gdal